util/rlp/src/impls.rs

```diff
@@ -0,0 +1,120 @@
+use byteorder::{WriteBytesExt, BigEndian};
+use bigint::prelude::{Uint, U128, U256, H64, H128, H160, H256, H512, H520, H2048};
+use traits::Encodable;
+use stream::RlpStream;
+
+impl Encodable for bool {
+	fn rlp_append(&self, s: &mut RlpStream) {
+		if *self {
+			s.encoder().encode_value(&[1]);
+		} else {
+			s.encoder().encode_value(&[0]);
+		}
+	}
+}
+
+impl<'a> Encodable for &'a [u8] {
+	fn rlp_append(&self, s: &mut RlpStream) {
+		s.encoder().encode_value(self);
+	}
+}
+
+impl Encodable for Vec<u8> {
+	fn rlp_append(&self, s: &mut RlpStream) {
+		s.encoder().encode_value(self);
+	}
+}
+
+impl<T> Encodable for Option<T> where T: Encodable {
```

Comment:
It's not a question to me. It's already like that on master,

https://github.com/ethcore/parity/blob/8a67a0a80aae506233f65cc41962fe483fc10c15/util/rlp/src/rlpstream.rs#L340-L350